include/deal.II/fe/fe_tools_extrapolate.templates.h

```diff
@@ -1164,94 +1183,92 @@ namespace FETools
 
     template <int dim, int spacedim, class OutVector>
     void
-    ExtrapolateImplementation<dim, spacedim, OutVector>::send_cells(
+    ExtrapolateImplementation<dim, spacedim, OutVector>::exchange_data_on_cells(
       const std::vector<CellData> &cells_to_send,
       std::vector<CellData> &      received_cells) const
     {
-      std::vector<std::vector<char>> sendbuffers(cells_to_send.size());
-      std::vector<MPI_Request>       requests(cells_to_send.size());
-      std::vector<unsigned int>      destinations;
-
-      // Protect the communication below:
-      static Utilities::MPI::CollectiveMutex      mutex;
-      Utilities::MPI::CollectiveMutex::ScopedLock lock(mutex, communicator);
-
-      // We pick a new tag in each round. Wrap around after 10 rounds:
-      const int mpi_tag =
-        Utilities::MPI::internal::Tags::fe_tools_extrapolate + round % 10;
-
-      // send data
-      std::vector<std::vector<char>>::iterator buffer = sendbuffers.begin();
-      unsigned int                             idx    = 0;
-      for (typename std::vector<CellData>::const_iterator it =
-             cells_to_send.begin();
-           it != cells_to_send.end();
-           ++it, ++idx)
-        {
-          destinations.push_back(it->receiver);
-
-          *buffer        = it->pack_data();
-          const int ierr = MPI_Isend(buffer->data(),
-                                     buffer->size(),
-                                     MPI_BYTE,
-                                     it->receiver,
-                                     mpi_tag,
-                                     communicator,
-                                     &requests[idx]);
-          AssertThrowMPI(ierr);
-
-          ++buffer;
-        }
+      // First figure out where we need to send stuff. Some of the cells
+      // in the input argument to this function might be destined for
+      // the same process, so we have to only look at the unique set of
+      // destinations:
+      std::set<types::subdomain_id> destinations;
+      for (const auto &cell : cells_to_send)
+        destinations.insert(cell.receiver);
```

Comment:
Let's use a vector here. We try to not use `std::set`. See the comment: https://github.com/dealii/dealii/blob/3447d0a9ff1042a7171ebe90c28a84c24d1be00c/source/dofs/dof_tools.cc#L1169-L1174

Reply:
Ah, interesting point! So done!